sinkingship.py

```diff
@@ -92,9 +93,16 @@ def iterator():
             "reblog_info": True,
             "notes_info": True,
         }
+        if before:
+            params["before"]=before
 
         while True:
-            resp = sess.get(api_url, params=params)
+            try:
+                resp = sess.get(api_url, params=params)
+            except:
+                print("Unable to download posts before %i."%params["before"])
+                print("Call this script again in a few hours with --before=%i to download older posts."%params["before"])
+                raise
```

Comment:
Eek! Bare `except` is rarely what you want – it catches very broadly, including KeyboardInterrupt and SystemExit. Better to scope it more tightly, and provide a comment explaining what condition you're trying to prevent.

Also, this `except` handler will throw a KeyError if it fails on the first iteration, because there won't be a `"before"` key in `params`.